src/sagemaker/clarify.py

```diff
@@ -393,6 +438,92 @@ def to_dict(self) -> Dict[str, Any]:  # pragma: no cover
+        item_id: Union[str, int],
+        timestamp: Union[str, int],
+        related_time_series: Optional[List[Union[str, int]]] = None,
+        item_metadata: Optional[List[Union[str, int]]] = None,
+    ):
+        """Initialises TimeSeries explainability data configuration fields.
+
+        Args:
+            target_time_series (str or int): A string or a zero-based integer index.
+                Used to locate the target time series in the shared input dataset.
+                If this parameter is a string, then all other parameters must also
+                be strings or lists of strings. If this parameter is an int, then
+                all others must be ints or lists of ints.
+            item_id (str or int): A string or a zero-based integer index. Used to
+                locate item id in the shared input dataset.
+            timestamp (str or int): A string or a zero-based integer index. Used to
+                locate timestamp in the shared input dataset.
+            related_time_series (list[str] or list[int]): Optional. An array of strings
+                or array of zero-based integer indices. Used to locate all related time
+                series in the shared input dataset (if present).
+            item_metadata (list[str] or list[int]): Optional.  An array of strings or
+                array of zero-based integer indices. Used to locate all item metadata
+                fields in the shared input dataset (if present).
+
+        Raises:
+            AssertionError: If any required arguments are not provided.
+            ValueError: If any provided arguments are the wrong type.
+        """
+        # check target_time_series, item_id, and timestamp are provided
+        assert target_time_series, "Please provide a target time series."
+        assert item_id, "Please provide an item id."
+        assert timestamp, "Please provide a timestamp."
+        # check all arguments are the right types
+        if not isinstance(target_time_series, (str, int)):
+            raise ValueError("Please provide a string or an int for ``target_time_series``")
+        params_type = type(target_time_series)
+        if not isinstance(item_id, params_type):
+            raise ValueError(f"Please provide {params_type} for ``item_id``")
+        if not isinstance(timestamp, params_type):
+            raise ValueError(f"Please provide {params_type} for ``timestamp``")
+        # add remaining fields to an internal dictionary
+        self.time_series_data_config = dict()
+        _set(target_time_series, "target_time_series", self.time_series_data_config)
+        _set(item_id, "item_id", self.time_series_data_config)
+        _set(timestamp, "timestamp", self.time_series_data_config)
+        # check optional arguments are right types if provided
+        related_time_series_error_message = (
+            f"Please provide a list of {params_type} for ``related_time_series``"
+        )
+        if related_time_series:
```

Comment:
Note to self: add validation step to check that if `related_time_series` or `item_metadata` are lists of JMESPaths (strings), then no entry should be blank strings.